### DISCUSSION TITLE
borks when trying to include a package in a temporal shell

### DISCUSSION THREAD

**Discussion by plchldr-jm** [nix-community/nix-on-droid#410] [Q&A]
```shell
Welcome to Nix-on-Droid!
If nothing works, open an issue at https://github.com/nix-community/nix-on-droid/issues or try the rescue shell.
-bash-5.2$ nix-shell -p cmatrix
warning: Nix search path entry '/data/data/com.termux.nix/files/home/.nix-defexpr/channels/nixpkgs/' does not exist, ignoring
error:
       … <borked>

         at «none»:0: (source not available)

       … while calling the 'import' builtin

         at «string»:1:18:

            1| {...}@args: with import <nixpkgs> args; (pkgs.runCommandCC or pkgs.runCommand) "shell" { buildInputs = [ (cmatrix) ]; } ""
             |                  ^

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: file 'nixpkgs' was not found in the Nix search path (add it using $NIX_PATH or -I)

       at «none»:0: (source not available)
-bash-5.2$
```
is there something i need to preconfig or something?

**Comment by t184256** [comment]
On August 30, 2024 9:25:20 PM GMT+02:00, Jim ***@***.***> wrote:
>```shell
>Welcome to Nix-on-Droid!
>If nothing works, open an issue at https://github.com/nix-community/nix-on-droid/issues or try the rescue shell.
>-bash-5.2$ nix-shell -p cmatrix
>warning: Nix search path entry '/data/data/com.termux.nix/files/home/.nix-defexpr/channels/nixpkgs/' does not exist, ignoring
>error:
>       … <borked>
>
>         at «none»:0: (source not available)
>
>       … while calling the 'import' builtin
>
>         at «string»:1:18:
>
>            1| ***@***.***: with import <nixpkgs> args; (pkgs.runCommandCC or pkgs.runCommand) "shell" { buildInputs = [ (cmatrix) ]; } ""
>             |                  ^
>
>       (stack trace truncated; use '--show-trace' to show the full trace)
>
>       error: file 'nixpkgs' was not found in the Nix search path (add it using $NIX_PATH or -I)
>
>       at «none»:0: (source not available)
>-bash-5.2$
>```
>is there something i need to preconfig or something?
>

Have you installed with flakes? If yes, you won't have a nixpkgs channel. Use `nix shell nixpkgs#cmatrix` instead.

**Comment by plchldr-jm** [comment]
yes, i've just tried flakes recently, sorry fot not pointing that one out

so, how long does it usually take to `evaluating derivation 'flake:nixpkgs#cmatrix'`?

**Comment by t184256** [comment]
Depends on the device performance, it's proot+storage-bottlenecked. I'd say, if it's >10 minutes worth power saving disabled and wakelock acquired from the notification, that's too much.

**Comment by plchldr-jm** [comment]
didn't see when it succeeded, so not sure how long, but it worked like a charm, thanks

**Comment by bohanubis** [comment]
```

    # pin the registry to avoid downloading and evaling a new nixpkgs version every time
    registry = lib.mapAttrs (_: v: {flake = v;}) inputs;

    # set the path for channels compat
    nixPath = lib.mapAttrsToList (key: _: "${key}=flake:${key}") config.nix.registry;
    
 ```
 
 can't these lines work ????
@t184256
